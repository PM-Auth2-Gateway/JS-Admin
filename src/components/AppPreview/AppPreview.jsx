import React from 'react';
import { Link } from 'react-router-dom';

import styles from './AppPreview.module.scss';

import appIcon from '../../assets/app-icon.svg';

const AppPreview = ({ url, name }) => {
  return (
    <div className={styles.appPreview}>
      <Link to={url}>
        <img src={appIcon} className={styles.appIcon} alt='icon' />
      </Link>
      <span className={styles.appName}>
        <Link to={url}>{name}</Link>
      </span>
    </div>
  );
};

export default AppPreview;
