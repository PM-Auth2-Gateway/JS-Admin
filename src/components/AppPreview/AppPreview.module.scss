.appPreview {
  display: flex;
  align-items: flex-start;
  gap: 16px;

  .appIcon {
    width: 40px;
    border-radius: 4px;
    background-color: #eff0f2;
  }

  .appName {
    font-size: 18px;

    a {
      color: #0A84AE;
      text-decoration: none;

      &:hover {
        color: #053B4E;
      }
    }
  }
}
