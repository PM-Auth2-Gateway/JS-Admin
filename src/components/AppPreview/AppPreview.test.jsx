import React from "react";
import {shallow} from 'enzyme';

import AppPreview from "./AppPreview";

jest.mock('react-router-dom', () => ({
  Link: "Link"
}))

describe("AppPreview Component", () => {
  it('should render AppPreview Component', function () {

    const component = shallow(<AppPreview url={'/application/1'} name={'App Name'}/>);

    expect(component).toMatchSnapshot();
  });
})
