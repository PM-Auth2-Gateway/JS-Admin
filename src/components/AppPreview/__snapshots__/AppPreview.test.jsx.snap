// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPreview Component should render AppPreview Component 1`] = `
<div
  className="appPreview"
>
  <Link
    to="/application/1"
  >
    <img
      alt="icon"
      className="appIcon"
      src="app-icon.svg"
    />
    <span
      className="appName"
    >
      App Name
    </span>
  </Link>
</div>
`;
